Startup crash if SMT solver timeout is increased (Integer cannot be cast to class Long)

## Description

If the (global) SMT solver timeout is set to 10 seconds, KeY crashes at startup. The default setting (2 seconds) works.

## Reproducible
  
always

### Steps to reproduce 

1. Open SMT Solver Settings
2. Set timeout to 10
3. Restart KeY

expected: works fine
actual: KeY crashes immediately

### Additional information

<details>
<summary>Stacktrace</summary>

```
211        INFO  main            d.u.i.k.c.Main            Assertion evaluation is enabled.
213        INFO  main            d.u.i.k.c.Main            Memory: total 2032.0 MB, max 30688.0 MB, free 2001.4916915893555 MB
Exception in thread "main" java.lang.ExceptionInInitializerError
        at de.uka.ilkd.key.core.Main.setEnabledExperimentalFeatures(Main.java:385)
        at de.uka.ilkd.key.core.Main.call(Main.java:262)
        at de.uka.ilkd.key.core.Main.call(Main.java:55)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at de.uka.ilkd.key.core.Main.main(Main.java:227)
Caused by: java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
        at de.uka.ilkd.key.settings.AbstractPropertiesSettings.lambda$createIntegerProperty$4(AbstractPropertiesSettings.java:138)
        at de.uka.ilkd.key.settings.AbstractPropertiesSettings$DefaultPropertyEntry.fromObject(AbstractPropertiesSettings.java:273)
        at de.uka.ilkd.key.settings.AbstractPropertiesSettings.lambda$readSettings$1(AbstractPropertiesSettings.java:114)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at de.uka.ilkd.key.settings.AbstractPropertiesSettings.readSettings(AbstractPropertiesSettings.java:113)
        at de.uka.ilkd.key.settings.ViewSettings.readSettings(ViewSettings.java:267)
        at de.uka.ilkd.key.settings.ProofIndependentSettings.load(ProofIndependentSettings.java:115)
        at de.uka.ilkd.key.settings.ProofIndependentSettings.loadSettings(ProofIndependentSettings.java:94)
        at de.uka.ilkd.key.settings.ProofIndependentSettings.<init>(ProofIndependentSettings.java:71)
        at de.uka.ilkd.key.settings.ProofIndependentSettings.<clinit>(ProofIndependentSettings.java:35)
        ... 12 more
```
</details>

---

* Commit: 0ded37352caf2fe8c6abf2495252f6125373ac71 (latest main)

--- a/key.core/src/main/java/de/uka/ilkd/key/settings/AbstractPropertiesSettings.java
+++ b/key.core/src/main/java/de/uka/ilkd/key/settings/AbstractPropertiesSettings.java
@@ -127,22 +127,24 @@ public abstract class AbstractPropertiesSettings extends AbstractSettings {
     }
 
     protected PropertyEntry<Double> createDoubleProperty(String key, double defValue) {
-        PropertyEntry<Double> pe =
-            new DefaultPropertyEntry<>(key, defValue, parseDouble, (it) -> (double) it);
+        PropertyEntry<Double> pe = new DefaultPropertyEntry<>(key, defValue, parseDouble,
+            (it) -> ((Number) it).doubleValue());
         propertyEntries.add(pe);
         return pe;
     }
 
     protected PropertyEntry<Integer> createIntegerProperty(String key, int defValue) {
+        // A stored numeric value may deserialize as Integer or Long depending on its magnitude and
+        // the settings format, so accept any Number rather than assuming a particular boxed type.
         PropertyEntry<Integer> pe = new DefaultPropertyEntry<>(key, defValue, parseInt,
-            (it) -> Math.toIntExact((Long) it));
+            (it) -> Math.toIntExact(((Number) it).longValue()));
         propertyEntries.add(pe);
         return pe;
     }
 
     protected PropertyEntry<Float> createFloatProperty(String key, float defValue) {
-        PropertyEntry<Float> pe =
-            new DefaultPropertyEntry<>(key, defValue, parseFloat, (it) -> (float) (double) it);
+        PropertyEntry<Float> pe = new DefaultPropertyEntry<>(key, defValue, parseFloat,
+            (it) -> ((Number) it).floatValue());
         propertyEntries.add(pe);
         return pe;
     }

--- a/key.core/src/test/java/de/uka/ilkd/key/settings/AbstractPropertiesSettingsTest.java
+++ b/key.core/src/test/java/de/uka/ilkd/key/settings/AbstractPropertiesSettingsTest.java
@@ -1,0 +1,45 @@
+/* This file is part of KeY - https://key-project.org
+ * KeY is licensed under the GNU General Public License Version 2
+ * SPDX-License-Identifier: GPL-2.0-only */
+package de.uka.ilkd.key.settings;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Regression tests for {@link AbstractPropertiesSettings} numeric properties: a stored value may
+ * deserialize as {@link Integer} or {@link Long} depending on its magnitude and the settings
+ * format,
+ * and reading it back must not depend on which boxed type it happens to be (it used to crash KeY on
+ * startup with a {@code ClassCastException}).
+ */
+public class AbstractPropertiesSettingsTest {
+
+    private static final class TestSettings extends AbstractPropertiesSettings {
+        final PropertyEntry<Integer> intProp;
+
+        TestSettings() {
+            super("TestCat");
+            intProp = createIntegerProperty("myInt", 7);
+        }
+    }
+
+    @Test
+    public void readsIntegerPropertyStoredAsInteger() {
+        Configuration cfg = new Configuration();
+        cfg.getOrCreateSection("TestCat").set("myInt", 42); // stored as Integer
+        TestSettings settings = new TestSettings();
+        settings.readSettings(cfg);
+        assertEquals(42, settings.intProp.get());
+    }
+
+    @Test
+    public void readsIntegerPropertyStoredAsLong() {
+        Configuration cfg = new Configuration();
+        cfg.getOrCreateSection("TestCat").set("myInt", Long.valueOf(42)); // stored as Long
+        TestSettings settings = new TestSettings();
+        settings.readSettings(cfg);
+        assertEquals(42, settings.intProp.get());
+    }
+}

--- a/key.ui/src/main/java/de/uka/ilkd/key/gui/RecentFileMenu.java
+++ b/key.ui/src/main/java/de/uka/ilkd/key/gui/RecentFileMenu.java
@@ -263,7 +263,11 @@ public class RecentFileMenu {
         public Configuration asConfiguration() {
             Configuration config = new Configuration();
             config.set(KEY_PATH, path);
-            config.set(KEY_PROFILE, (Object) profile);
+            // Only store a real profile name; writing a null serializes to the string "null", which
+            // then fails to resolve on reload.
+            if (profile != null) {
+                config.set(KEY_PROFILE, profile);
+            }
             config.set(KEY_LOAD_SINGLE_JAVA, singleJava);
             config.set(KEY_OPTIONS, additionalOption);
             return config;
@@ -305,15 +309,18 @@ public class RecentFileMenu {
                 }
             } else {
                 String profileName = fileEntry.profile;
-                var selectedProfile =
-                    ServiceLoader.load(DefaultProfileResolver.class)
-                            .stream()
-                            .filter(it -> it.get().getProfileName().equals(profileName))
-                            .findFirst()
-                            .map(it -> it.get().getDefaultProfile());
+                // A missing profile -- null, or the literal string "null" that older recent-file
+                // entries stored for it -- means "use the default profile", not an error.
+                boolean hasProfile = profileName != null && !profileName.equals("null");
+                var selectedProfile = hasProfile
+                        ? ServiceLoader.load(DefaultProfileResolver.class)
+                                .stream()
+                                .filter(it -> it.get().getProfileName().equals(profileName))
+                                .findFirst()
+                                .map(it -> it.get().getDefaultProfile())
+                        : Optional.<Profile>empty();
 
-
-                if (profileName != null && selectedProfile.isEmpty()) {
+                if (hasProfile && selectedProfile.isEmpty()) {
                     JOptionPane.showMessageDialog(mainWindow,
                         "Could not find previous selected profile %s.".formatted(profileName));
                     return;

--- a/key.ui/src/main/java/de/uka/ilkd/key/gui/WindowUserInterfaceControl.java
+++ b/key.ui/src/main/java/de/uka/ilkd/key/gui/WindowUserInterfaceControl.java
@@ -210,6 +210,10 @@ public class WindowUserInterfaceControl extends AbstractMediatorUserInterfaceCon
             if (!isAtLeastOneMacroRunning()) {
                 mainWindow.hideStatusProgress();
                 assert info instanceof ProofMacroFinishedInfo;
+                // Show the macro's aggregate result (total rules applied / goals closed). Without
+                // this the status line keeps whatever the macro's last internal strategy run left
+                // there -- a tiny partial count rather than the whole macro's work.
+                mainWindow.displayResults(info.toString());
                 final Proof proof = (Proof) info.getProof();
                 if (proof != null && !proof.closed()
                         && mainWindow.getMediator().getSelectedProof() == proof) {

